pkg/cmd/template/cmd_test.go

```diff
@@ -826,3 +831,513 @@ func TestLoadYTTModuleFailEarly(t *testing.T) {
 		t.Fatalf("Expected RunWithFiles to fail with error, but was '%s'", out.Err.Error())
 	}
 }
+
+func TestNonYamlFilesSuggestsUsingOutputFlags(t *testing.T) {
```

Comment:
if we are going to have top level test for this feature, we should do something similar to this: https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/cmd/template/cmd_test.go#L15. you may have to bubble up to testing against Run() function and in that case you should inject your own ui as interface instead of trying to capture os.* fds.

Reply:
@pivotal-dean and I reviewed the tests and decided to reduce the amount of tests included. We decided to keep testing at the command level versus using `Run()` because it is easier to understand as far as what actual command is being run and the expected output.

We still need to use the redirecting of stderr/out because the stderr is only written via `ui.Warnf` and no error is ever returned from `Output()`.

We were able to refactor the test to focus on `Output()` for the main test in regular_input_test.go, but the higher level command tests we thought were valuable for including tests with `--output-files`.